src/Nest/Aggregations/AggregateFormatter.cs

```diff
@@ -889,9 +889,9 @@ private IBucket GetCompositeBucket(ref JsonReader reader, IJsonFormatterResolver
 			else
 				reader.ReadNext(); // }
 
-			var significantTermItem = new SignificantTermsBucket(subAggregates)
+			var significantTermItem = new SignificantTermsBucket<object>(subAggregates)
 			{
-				Key = (string)key,
```

Comment:
key is known to be only `string` or `double` from where its called:

```csharp
		private IBucket GetKeyedBucket(ref JsonReader reader, IJsonFormatterResolver formatterResolver)
```


 perhaps we can use this to our advantage and instantiate either a `SignificantTermBucket<string>` or `SignificantTermBucket<double>` and then use this type information in 

```csharp
private IEnumerable<SignificantTermsBucket<TKey>> GetSignificantTermsBuckets<TKey>(IEnumerable<IBucket> items)	
```

To return the `items` (casted) directly instead.

Reply:
You could do the same thing in `GetKeyedBucket`. Should that one be refactored as well?